Fix nix-shell and nix-build

nc6 opened the pull request:
- Alters the nix configuration so as to actually make use of the
  separate hackage.nix pin.
- Update the hackage.nix and haskell.nix versions sufficiently for the
  new index-state. We bump haskell.nix only to the same version as
  cardano-node, to take advantage of the cache.

nc6 committed:
This commit does two things:

- Alters the nix configuration so as to actually make use of the
  separate hackage.nix pin.
- Update the hackage.nix and haskell.nix versions sufficiently for the
  new index-state. We bump haskell.nix only to the same version as
  cardano-node, to take advantage of the cache.

lehins commented:
@nc6 Since you updating nix config can you also update GHC to 8.10.7 in this PR?

https://github.com/input-output-hk/cardano-ledger/blob/7155301bcddc9cc2efe9cd98ad9c2521e9a0d085/nix/haskell.nix#L11

nc6 commented:
I can, though this is ultimately something we should coordinate across the various components.

nc6 committed:
Update to GHC8.10.7

JaredCorduan reviewed (approved):
thanks @nc6 !

nc6 committed:
Update ShelleyGenesis golden test.

This is a result of a bump in Aeson versions and how they internally
represent records. No actual data has changed, just ordering.
